packages/ember-model/lib/model.js

```diff
@@ -398,6 +398,32 @@ Ember.Model = Ember.Object.extend(Ember.Evented, {
 Ember.Model.reopenClass({
   primaryKey: 'id',
 
+  create: function(properties) {
+    var transformedProperties = {};
+    if (properties) {
+      var attributes = this.getAttributes();
+
+      for (var i=0; i<attributes.length; i++) {
+        var attribute = attributes[i];
+        var transformedKey;
+        var meta = this.metaForProperty(attribute);
+
+        if (meta.options && meta.options.key) {
+          transformedKey = this.camelizeKeys ? underscore(meta.options.key) : meta.options.key;
+        } else {
+          transformedKey = this.camelizeKeys ? underscore(attribute) : attribute;
+        }
+
+        transformedProperties[transformedKey] = properties[attribute];
+      }
+    }
+
+    var obj = this._super.apply(this, arguments);
+    obj.set('_data', transformedProperties);
+    obj.setProperties(properties);
```

Comment:
Rather than transforming the properties here before we have an instance l, could we instead just call _super and then .serialize each of the .dirtyattributes and set the result as _data? That might avoid the duplication

Reply:
I don't quite follow. The way it's implemented, dirty tracking doesn't seem to work until we have set `_data`. I very well may be missing something though. If you think your alternative might be a valid approach, perhaps you could try it in a branch?

Reply:
Looks like there's already a method that nearly does what we need (`_copyDirtyAttributesToData`). I've pushed a commit here, but I'm not sure about it...

https://github.com/ebryn/ember-model/commit/643df6d67e07141f22002bd8804f50f2aa8c11f0

Reply:
Nice one, seems like a massive simplification. What don't you like about it?

Reply:
<img width="654" alt="Screen Shot 2019-04-10 at 14 25 10" src="https://user-images.githubusercontent.com/2526/55882459-7a83e200-5b9c-11e9-8dcd-0151d9e26cba.png">